webui/react/src/stores/resourcePools.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, PropsWithChildren, useCallback, useContext, useState } from 'react';
+
+import { getResourcePools } from 'services/api';
+import { ResourcePool } from 'types';
+import handleError from 'utils/error';
+import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+
+type ResourcePoolsContext = {
+  resourcePools: Loadable<ResourcePool[]>;
+  updateResourcePools: (
+    fn: (resourcePools: Loadable<ResourcePool[]>) => Loadable<ResourcePool[]>,
+  ) => void;
+};
+
+const ResourcePoolsContext = createContext<ResourcePoolsContext | null>(null);
+
+export const ResourcePoolsProvider: React.FC<PropsWithChildren> = ({ children }) => {
+  const [state, setState] = useState<Loadable<ResourcePool[]>>(NotLoaded);
+  return (
+    <ResourcePoolsContext.Provider value={{ resourcePools: state, updateResourcePools: setState }}>
+      {children}
+    </ResourcePoolsContext.Provider>
+  );
+};
+
+export const useFetchResourcePools = (canceler: AbortController): (() => Promise<void>) => {
+  const context = useContext(ResourcePoolsContext);
+  if (context === null) {
+    throw new Error('Attempted to use useFetchResourcePools outside of ResourcePool Context');
+  }
+  const { updateResourcePools } = context;
+
+  return useCallback(async (): Promise<void> => {
+    try {
+      const response = await getResourcePools({}, { signal: canceler.signal });
+      updateResourcePools(() => Loaded(response));
+    } catch (e) {
+      handleError(e);
+    }
+  }, [canceler, updateResourcePools]);
+};
+
+export const useResourcePools = (): ResourcePoolsContext => {
```

Comment:
The only place we're using `updateResourcePools` is a test. We should just return the resource pools here and use a mock for that test.

Reply:
resource pools are managed by the core platform as far as I know so we only ever want to know what the state of them is. If we do want some kind of update in the future though we'd want to design the API and not just expose the underlying update function here. (i.e. have add, remove, and update functions).